### PR TITLE
[Feature][HbaseSink]support array data.

### PR DESCRIPTION
**Pull request by lightzhao**
### Purpose of this pull request

#6052 hbase sink support array data.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java [hunk 1]
```diff
@@ -143,4 +146,32 @@ public void testHbaseSink(TestContainer container) throws IOException, Interrupt
...
+    public void testHbaseSinkWithArray(TestContainer container)
+            throws IOException, InterruptedException {
+        Container.ExecResult execResult = container.executeJob("/fake-to-hbase-array.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+        Table hbaseTable = hbaseConnection.getTable(table);
+        Scan scan = new Scan();
+        ArrayList<Result> results = new ArrayList<>();
+        ResultScanner scanner = hbaseTable.getScanner(scan);
+        for (Result result : scanner) {
+            System.out.println("Row Key: " + Bytes.toString(result.getRow()));
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java [hunk 2]
```diff
@@ -143,4 +146,32 @@ public void testHbaseSink(TestContainer container) throws IOException, Interrupt
...
+                String columnFamily = Bytes.toString(CellUtil.cloneFamily(cell));
+                String columnName = Bytes.toString(CellUtil.cloneQualifier(cell));
+                String value = Bytes.toString(CellUtil.cloneValue(cell));
+                System.out.println(
+                        "Column Family: "
+                                + columnFamily
+                                + ", Column Name: "
+                                + columnName
+                                + ", Value: "
+                                + value);
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]:
```suggestion
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java [hunk 3]
```diff
@@ -143,4 +146,32 @@ public void testHbaseSink(TestContainer container) throws IOException, Interrupt
...
+            }
+            results.add(result);
+        }
+        Assertions.assertEquals(results.size(), 5);
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: Please assert result value. Specially for array value.

Comment (lightzhao) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: done.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java (hunk 1 above)
Comment (lightzhao) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: done.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java (hunk 2 above)
Comment (lightzhao) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java]: done.

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**